TelepresenceBot/server-unit/core/server.js

```diff
@@ -1,21 +1,64 @@
 var io = require('socket.io').listen(5000);
+var ClientType = require("./clientType.js");
+var LoggingClient = require("./loggingClient.js");
 
-var clients = {};
+var room = "London";
+var humans = [];
+var bots = [];
+var testClient = new LoggingClient();
+
+io.use(function(client, next){
+    var rawClientType = client.handshake.query.clientType;
+    var clientType = ClientType.from(rawClientType);
+
+    switch(clientType) {
+        case ClientType.TEST:
+        case ClientType.BOT:
+            return next();
+        case ClientType.HUMAN:
+            var roomRoster = io.sockets.adapter.rooms[room];
+            if(roomRoster != undefined && roomRoster.length == 1) {
+                return next();
+            } else {
+                return next(new Error('No bots available'));
+            }
+        default:
+            return next(new Error('Unrecognised clientType: ' + rawClientType));
+
+    }
+});
 
 io.sockets.on('connection', function (client) {
 
-    console.log('connecting: ' + client.id);
-    clients[client.id] = client;
-    io.sockets.emit('connected', toKeysArrayFrom(clients));
+    var rawClientType = client.handshake.query.clientType;
+    var clientType = ClientType.from(rawClientType);
+
+    switch(clientType) {
+        case ClientType.HUMAN:
```

Comment:
`Human`s and `Bot`s get added to their respective arrays and then join the `London` room.